core/src/main/scala/no/nrk/bigquery/BQValueHasher.scala

```diff
@@ -0,0 +1,19 @@
+package no.nrk.bigquery
+
+import no.nrk.bigquery.syntax.bqShowInterpolator
+import scodec.bits.ByteVector
+
+import java.nio.charset.StandardCharsets
+
+trait BQValueHasher[T] {
+  def hashValueInBQ(value: Ident, range: BQRange): BQSqlFrag =
+    bqsql"""MOD(CAST(CONCAT("0x", SUBSTRING(TO_HEX(SHA256($value)), 0, 14)) as INT64), ${range.end})"""
+  def hashValue(value: T, range: BQRange): Long
```

Comment:
This might be a useful thing later:
```scala
def contramap[X](f: X => T): BQValueHasher[X] = {
  val self = this
  new BQValueHasher[X] {
    def hashvalue(value: X, range: BQRange) = self.hashValue(f(value), range)
  }
}
```